extensions/s3/src/main/java/io/deephaven/extensions/s3/S3CompletableOutputStream.java

```diff
@@ -284,6 +284,7 @@ private String initiateMultipartUpload() throws IOException {
         try {
             response = future.get();
         } catch (final InterruptedException | ExecutionException e) {
+            failAll(new IOException("Error initiating multipart upload for uri " + uri, e));
```

Comment:
We should not need to be failing by catching a get(); this should be a listener based on .whenCompleete or .exceptionally.